x-pack/legacy/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client_wrapper.js

```diff
@@ -93,26 +103,30 @@ export class SecureSavedObjectsClientWrapper {
     await this._ensureAuthorized(
       type,
       'update',
+      options.namespace,
       { type, id, attributes, options },
     );
 
     return await this._baseClient.update(type, id, attributes, options);
   }
 
-  async _checkSavedObjectPrivileges(actions) {
+  async _checkSavedObjectPrivileges(actions, namespace) {
```

Comment:
Until now, the [tests for this class](https://github.com/elastic/kibana/blob/7ac8e4d9ccc2a9e064d8a5621cd62453d1006164/x-pack/legacy/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client_wrapper.test.js) weren't concerned with testing the "spaces enabled" scenarios because that was covered by the [`checkPrivilegesDynamically` unit tests](https://github.com/elastic/kibana/blob/94c3cb671947df421a28602af484f5aa6c291721/x-pack/legacy/plugins/security/server/lib/authorization/check_privileges_dynamically.test.ts).

Now that we aren't using `checkPrivilegesDynamicallyWithRequest`, we end up with a testing gap. We can of course add tests, but the tests for this wrapper are already pretty lengthy and verbose, and I'm concerned about maintainability going forward if we add additional cases for the spaces enabled scenarios.

We could "spot check" one of the operations (`get`, for example) to make sure it works in the space-aware scenario, but I don't know how much that actually buys us...

Reply:
What if we introduced a `CheckSavedObjectPrivileges` "service" which is initialized and then passed to the `SecureSavedObjectsClientWrapper` in the constructor? That way we can mock out this dependency for the sake of the `SecureSavedObjectsClientWrapper` tests, and then test the permutations in the `CheckSavedObjectPrivileges` test themselves?

Reply:
Yep, I like that idea. My sleep deprived brain couldn't come to that conclusion today, thanks for the nudge 😄